File: neoforge/src/main/java/cc/turtl/turtlshell/neoforge/TurtlShellNeoForge.java
package cc.turtl.turtlshell.neoforge;

import cc.turtl.turtlshell.TurtlShell;
import cc.turtl.turtlshell.TurtlShellConstants;
import cc.turtl.turtlshell.config.TurtlShellConfig;
import net.minecraft.client.gui.screens.Screen;
import net.neoforged.fml.ModContainer;
import net.neoforged.fml.ModList;
import net.neoforged.fml.common.Mod;
import net.neoforged.neoforge.client.gui.IConfigScreenFactory;

@Mod(TurtlShellConstants.MOD_ID)
public final class TurtlShellNeoForge {
    public TurtlShellNeoForge() {
        registerConfigScreen();
        TurtlShell.initClient();
    }

    private void registerConfigScreen() {
        ModList.get().getModContainerById(TurtlShellConstants.MOD_ID)
                .ifPresent(c -> c.registerExtensionPoint(IConfigScreenFactory.class, this::createConfigScreen));
    }

    private Screen createConfigScreen(ModContainer container, Screen parent) {
        return TurtlShellConfig.createScreen(parent);
    }
}

File: neoforge/src/main/kotlin/cc/turtl/turtlshell/TurtlShellNeoForge.kt
package cc.turtl.turtlshell

import net.neoforged.bus.api.SubscribeEvent
import net.neoforged.fml.common.EventBusSubscriber
import net.neoforged.fml.common.Mod
import net.neoforged.neoforge.event.RegisterCommandsEvent

@EventBusSubscriber
@Mod(BuildDetails.MOD_ID)
object TurtlShellNeoForge {
    init {
        TurtlShell
    }

    @SubscribeEvent
    fun registerCommands(e: RegisterCommandsEvent) {
        TurtlShellCommands.register(e.dispatcher)
    }
}